### ISSUE TITLE
"bower list -p" should return packages ordered by dependency tree level

### ISSUE THREAD

**Issue by rassie** [bower/bower#1044]
Let's make a simple test first:

```
$ bower install d3-tip
....
$ bower list
└─┬ d3-tip#0.6.3 extraneous
  └── d3#3.4.0

$ bower list -p
{
  "d3-tip": "bower_components/d3-tip/index.js",
  "d3": "bower_components/d3/d3.js"
}
```

Now the problem is that `d3-tip` depends on `d3`, which is why `d3` has been installed at all. However, if any tool would like to re-use the output of `bower list -p` (or its API variation), the dependencies are skewed and their order is incorrect.

For example: I would like to split my `browserify` bundles in `common.js` and `app.js`, like in the [`bower-resolve` example](https://github.com/eugeneware/bower-resolve). I would iterate over `bower.commands.list()` output, resolve packages with `bower-resolve` and then add their filenames to the common bundle. However, if I just take the order provided by `Object.forEach` (most probably insertion order, however not guaranteed by standards), my common bundle would first feature `d3-tip` and only then `d3`, which is incorrect, since `d3` is a dependency of `d3-tip` and the latter expects the former to be loaded when called.

So in short, this is kind of messy, there needs to be a better way to get a linear dependency order from `bower`.
